### PR TITLE
Some fixes for bugs I encountered when trying to use this plugin.

### PR DESCRIPTION
**Pull request by Cyriuz**
See commit for details.

If you don't agree with the default stemmer option I added I can make a new PR without it.

### CODE REVIEW

File: src/server/utils/parseDocument.ts [hunk 1]
```diff
@@ -39,13 +39,21 @@ export function parseDocument($: cheerio.Root): ParsedDocument {
...
+            .not("header") // div.markdown
+            .children() // h1, p, p, h2, ...
+            .first(); // h1 || p
+          if ($firstElement.filter(HEADINGS).length > 0) {
+            return;
+          }
+        } else {
+          $("article > .markdown")
+            .children() // h1, p, p, h2, ...
+            .first(); // h1 || p
```
Comment (csantos1113) [inline src/server/utils/parseDocument.ts:56]: I think these lines are doing nothing?
maybe this should be assigned to `$firstElement`?

```suggestion
          $firstElement = $("article > .markdown")
            .children() // h1, p, p, h2, ...
            .first(); // h1 || p
```

Comment (Cyriuz) [reply src/server/utils/parseDocument.ts:56]: You're right! I added this branch to handle the tests for the older layout but I thought I added a test for this case too. I will investigate why the tests passes with this.

**Comment by weareoutman** [comment]
Hey @Cyriuz Thanks for your work, 👍

Could you follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), so we can let our release bot to generate change log automatically, like: `feat: add an option to ...` `fix: ...`. Thanks.

**Comment by weareoutman** [comment]
And next time, we would better to make different PRs for non-relevant features/fixes. Thus we can launch them independently. Like here we seem to get stuck with one fix, while other fixes seem to be good, but they can't be merged.

File: src/server/utils/parseDocument.ts (hunk 1 above)
Comment (Cyriuz) [reply src/server/utils/parseDocument.ts]: I misunderstood this code a bit, I think this is not needed now when everything is the same for the main heading. Also added a proper test for it this time :)

**Comment by Cyriuz** [comment]
Yeah sure, I can fixup the commit messages, thanks for the link! And yeah you're right, I was a bit lazy with this one, I'll split them in the future :)

**Review by weareoutman — APPROVED** [review]

**Comment by weareoutman** [comment]
Thanks, released in v0.24.0. By the way I just manually created a commit to mark these release notes.

**Comment by weareoutman** [comment]
There is a quick fix for indexing content between h1 and h2, and been released in v0.24.0.

**Pull request merged**